### DISCUSSION TITLE
How can I use generator in a viewmodel that modifies a model class ?

### DISCUSSION THREAD

**Discussion by rgranatagreisch** [CommunityToolkit/dotnet#116] [General]
Hello !

I am just starting an internship, and I have been tasked with building a .NET 5 - WPF application. We have decided that it should use the MVVM pattern, and one of the programmers advised me to have a look at "Source Generators". It does, indeed, looks very promising ! 

As I was planning to use the MVVM toolkit in my app anyway, I decided to try out the generator. But I am facing some issues, the first one being the hard-to-find documentation on the subject. I do understand it is a relatively new project, and I guess this issue will be gone as the project evolves. 

My second problem is, thus, linked to the first one. I have always made my viewmodels as "model wrappers", meaning that I always have a private field that is an instance of the model, and then my public properties are used to set / get the model fields. This allows me to keep my model "easily savable" and my viewmodel "easily displayable". But I have no idea how to make such a thing with Source Generation. Is it possible ? Is it even considered a good practice ? 

Any help would be greatly appreciated ! 

Thanks !

**Comment by Arlodotexe** [comment]
@rgranatagreisch I'm transferring this to the .NET Community Toolkit where we can better answer questions related to the MVVM Toolkit.

**Reply by rgranatagreisch** [reply]
Ok, thanks !

**Reply by michael-hawker** [reply]
The source generators are still new, most info about them can be found on their [issue here](https://github.com/CommunityToolkit/dotnet/issues/8).

I find if your Model objects already implement INotifyPropertyChange, then I'll just expose the model or collection of models directly to my view layer and bind to them directly. Re-wrapping each property of a model in a ViewModel for encapsulation is just a bunch of extra work with little benefit unless you're trying to coerce values and do extra business logic outside of commands for modifying their state.
